src/CMakeLists.txt

```diff
@@ -521,15 +521,21 @@ if(${TRITON_ENABLE_TRACING})
   if (NOT WIN32)
     target_include_directories(
       tracing-library
-      PRIVATE ${OPENTELEMETRY_CPP_INCLUDE_DIRS}
+      PUBLIC ${OPENTELEMETRY_CPP_INCLUDE_DIRS}
     )
 
     target_link_libraries(
       tracing-library
-      PRIVATE
+      PUBLIC
       ${OPENTELEMETRY_CPP_LIBRARIES})
   endif()
 
+  set_target_properties(
```

Comment:
Is this modification still required?

Reply:
Without the `-fPIC` flag set to ON, the CMake build process errors out.

Reply:
Here is a copy of the error message I received when building without :
![image](https://github.com/user-attachments/assets/60a726c6-1b15-4586-bc6f-3a9ecc51e85f)

Reply:
Wouldn't hurt to compile with position independent code.. But have you ever looked into compiling the binding as static library? It was mentioned in the design doc right? Was it not feasible?

Reply:
Initially attempted to compiling the binding as a static library as outlined in the design doc ([link](https://github.com/pybind/cmake_example/issues/11#issuecomment-405092832)):
```
add_library(supporting_library STATIC ...)
set_target_properties(supporting_library PROPERTIES POSITION_INDEPENDENT_CODE ON)
...
pybind11_add_module(pymodule MODULE ...)
target_link_libraries(pymodule PRIVATE supporting_library)
```
My initial approach was to link the dependencies(`http-endpoint-library` and `grpc-endpoint-libarry` to the `supporting_library`) and follow a similar procedure, but it resulted in symbol not found errors from the `pymodule` library.

Reply:
What symbols are not found..

Reply:
Apologies for not providing enough details. IIRC the symbol not found errors would be for a mangled function name within `triton::server::HTTPServer` / `triton::server::HTTPAPIServer`.

Reply:
@KrishnanPrash please try the relevant build and share the detailed errors with specific symbols - this will help the reviewers better understand the problems and decisions made